core-java-modules/core-java-lang-5/src/main/java/com/baeldung/generics/Cat.java

```diff
@@ -0,0 +1,30 @@
+package com.baeldung.generics;
+
+class Cat extends Animal implements Comparable<Cat> {
+    public Cat(String type, String name) {
+        super(type, name);
+    }
+
+    @Override
+    public String makeSound() {
+        return "Meow";
+    }
+
+    @Override
+    public int compareTo(Cat cat) {
+        return this.getName().length() - cat.getName().length();
+    }
```

Comment:
Since we are overriding the compareTo method we should also override the equals method making two animals that have the same name length equal. In this way we would comply with the recommendation:
`x.compareTo(y)==0) == (x.equals(y)`

Reply:
Thank you @JavierPeris, I've updated the code.

Reply:
Hi @apeterlic the equals itself doesn't have an impact on the article. However, please pay attention to the rule:
x.compareTo(y)==0) == (x.equals(y) whenever compareTo returns a 0, the equals have to return true with your definition of comparing an equals this condition is not met. It's OK as long as the consumer of the code is warned due to the impact it can have.

Reply:
Hi @JavierPeris I understand. Would it be ok if we just add comment? Otherwise, I'd need to modify compareTo method.

Reply:
@apeterlic IIt's OK with a comment.

Reply:
@JavierPeris, thanks. I added a comment.